tao_compiler/mlir/disc/transforms/disc_memref_gvn.cc

```diff
@@ -0,0 +1,118 @@
+// Copyright 2022 The BladeDISC Authors. All rights reserved.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// This file implements GVN optimization on memref dialect.
```

Comment:
I agree the current GVN implement is limited. But we do have value numbering function in this pass. I name the entire pass as GVN because we may add more GVN functions in this pass in the future.

Reply:
TBD.